pkg/runner/runner.go

```diff
@@ -126,8 +139,24 @@ func (runner *ZeroRunner) initializeMainEntrypoint() (mem.MemoryAddress, error)
 	if err != nil {
 		return mem.UnknownAddress, err
 	}
-
-	if runner.proofmode {
+	switch runner.runnerMode {
+	case ExecutionMode:
+		returnFp := memory.AllocateEmptySegment()
+		mvReturnFp := mem.MemoryValueFromMemoryAddress(&returnFp)
+		mainPCOffset, ok := runner.program.Entrypoints["main"]
+		if !ok {
+			return mem.UnknownAddress, errors.New("can't find an entrypoint for main")
+		}
+		return runner.initializeEntrypoint(mainPCOffset, nil, &mvReturnFp, memory, stack, 0)
+	case ProofModeCairo1:
+		returnFp := memory.AllocateEmptySegment()
+		mvReturnFp := mem.MemoryValueFromMemoryAddress(&returnFp)
+		mainPCOffset, ok := runner.program.Entrypoints["main"]
+		if !ok {
+			return mem.UnknownAddress, errors.New("can't find an entrypoint for main")
+		}
+		return runner.initializeEntrypoint(mainPCOffset, nil, &mvReturnFp, memory, stack, 2)
```

Comment:
It's the same logic for both I see except for the `cairo1FpOffset` you are passing to initializeEntrypoint function, am I correct? Would it be better to unify these two and pass `cairo1FpOffset` as a parameter instead of the direct value?

Reply:
Yeah, thats good idea 💡